### PR TITLE
docs: Add how to release doc

### PR DESCRIPTION
**Pull request by jimmidyson**
**What problem does this PR solve?**:
Adds docs with details how to release.

**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

**Review by dkoshkin — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]
Nice!
```
➜  capi-runtime-extensions git:(jimmi/releasing-doc) ✗ which release-please
/Users/dimitri.koshkin/src/github.com/d2iq-labs/capi-runtime-extensions/.devbox/nix/profile/default/bin/release-please
➜  capi-runtime-extensions git:(jimmi/releasing-doc) ✗ release-please --version
16.10.2
```

File: hack/flakes/flake.nix [hunk 1]
```diff
@@ -81,6 +81,19 @@
               "-X" "sigs.k8s.io/cluster-api-provider-aws/v2/version.gitTreeState=clean"
             ];
           };
+
+          release-please = buildNpmPackage rec {
```
Comment (faiq) [inline hack/flakes/flake.nix:85]: nice

File: docs/content/contributing/releasing.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
++++
+title = "Releasing"
++++
+
+This project uses [release-please] to automate changelog updates per release. Due to security restrictions[^1] in the
+`nutanix-cloud-native` GitHub organisation, the release process is a little more complex than just using the
```
Comment (faiq) [inline docs/content/contributing/releasing.md:6]:
```suggestion
`nutanix-cloud-native` GitHub organization, the release process is a little more complex than just using the
```

File: docs/content/contributing/releasing.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[release-please-action].
+
+When a release has been cut, a new release PR can be created manually using the `release-please` CLI locally. This needs
+to be run by someone with write permissions to the repository. Create the `release-please` branch and PR:
+
+```bash
+release-please release-pr --repo-url nutanix-cloud-native/cluster-api-runtime-extension --token "$(gh auth token)"
```
Comment (faiq) [inline docs/content/contributing/releasing.md:13]: Why not use make targets?

Comment (jimmidyson) [reply docs/content/contributing/releasing.md:13]: Good point! Let me add that.

File: docs/content/contributing/releasing.md (hunk 2 above)
Comment (jimmidyson) [reply docs/content/contributing/releasing.md:6]: It's wrong, but I'll allow it.

File: docs/content/contributing/releasing.md (hunk 3 above)
Comment (jimmidyson) [reply docs/content/contributing/releasing.md]: Done.

**Review by faiq — APPROVED** [review]

**Pull request merged**